### PR TITLE
RAG enhancements 

### PR DESCRIPTION
**Pull request by spuchol81**
Same purpose than the previous one, but with more accuracy. The model is constrained to use the recipe found in the rag. Obtained much more consistent result compared to QuestionAnswerAdvisor.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,3 +39,8 @@ out/
 ### VS Code ###
 .vscode/
 .env
+
+
+# Exclude creds.yaml
+creds.yaml
+.DS_Store
```
Comment (SandraAhlgrimm) [inline .gitignore:46]: .DS_Store is already in line 8

Where is the creds.yaml coming from?

File: src/main/java/com/example/RecipeService.java [hunk 2]
```diff
@@ -48,7 +58,15 @@ public RecipeService(ChatClient chatClient, Optional<ImageModel> imageModel, Vec
 
     public void addRecipeDocumentForRag(Resource pdfResource) {
         log.info("Add recipe document {} for rag", pdfResource.getFilename());
-        var documentReader = new PagePdfDocumentReader(pdfResource);
+        var documentReader = new PagePdfDocumentReader(pdfResource,
+        PdfDocumentReaderConfig.builder()
```
Comment (SandraAhlgrimm) [inline src/main/java/com/example/RecipeService.java:62]: thank you, that's great!

File: src/main/resources/application-openai.yaml [hunk 3]
```diff
@@ -6,5 +6,4 @@ spring:
     chat:
       options.temperature: 1.0
     image.options:
-      model: dall-e-3
-    api-key:
```
Comment (SandraAhlgrimm) [inline src/main/resources/application-openai.yaml:10]: I'd keep the api-key here to make it obvious that it is needed to use OpenAI

**Review by SandraAhlgrimm — CHANGES_REQUESTED** [review]
Thank you! I have a few things commented. And would you like to update the README as well?

**Comment by timosalm** [comment]
Thanks for the additional input @SandraAhlgrimm. 
I'm currently working on enhancements for this PR.

File: .gitignore (hunk 1 above)
Comment (spuchol81) [reply .gitignore:46]: Hello Sandra, I used to include API key into app properties from separate file on another project but it didn't work this time :(

**Pull request merged**